test/framework/alertmanager.go

```diff
@@ -398,6 +400,49 @@ func (f *Framework) GetSilences(ctx context.Context, ns, n string) (models.Getta
 	return getSilencesResponse, nil
 }
 
+func (f *Framework) WaitForAlertmangerFiringAlert(ctx context.Context, ns, svcName, alertName string) error {
+	err := wait.Poll(10*time.Second, time.Minute*5, func() (bool, error) {
+		alerts, err := f.GetAlerts(ctx, ns, svcName)
+		if err != nil {
+			return false, err
+		}
+
+		for _, alert := range alerts {
+			if alert.Labels["alertname"] == alertName && alert.Status.State != pointer.String("firing") {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to wait for alert %s to fire: %v", alertName, err)
+	}
+
+	return nil
+}
+
+func (f *Framework) GetAlerts(ctx context.Context, ns, svcName string) (models.GettableAlerts, error) {
+	apiResponse := models.GettableAlerts{}
+
+	resp, err := f.AlertmangerSVCGetRequest(ctx, ns, svcName, "/api/v2/alerts", nil)
```

Comment:
We could narrow down the request to return only the expected alert.
```suggestion
	resp, err := f.AlertmangerSVCGetRequest(ctx, ns, svcName, "/api/v2/alerts?state=active&filter=alertname="+alertName, nil)
```

Reply:
The query parameters have to go on the front otherwise we get an error that the Service was not found
```golang
		resp, err := f.AlertmangerSVCGetRequest(ctx, ns, svcName, "/api/v2/alerts", map[string]string{
			"state":  "active",
			"filter": "alertname=" + alertName,
		})
```